### PR TITLE
FACES-2656 alloy:accordion/alloy:tabView client-side selected index does not match server-side selected index (extract data rendering code to separate DataEncoderBase class)

### PR DESCRIPTION
**Pull request by stiemannkj1**
@ngriffin7a, please merge and backport these **two** commits.

### CODE REVIEW

**Comment by ngriffin7a** [comment]
@stiemannkj1 This branch has a merge conflict. Please fix one or both of the commits and let me know when it is ready again. Thanks.

**Comment by stiemannkj1** [comment]
@ngriffin7a, updated, retested, and ready to be merged (**two commits**).

File: alloy/src/main/resources/META-INF/resources/liferay-faces-alloy/alloy.js [hunk 1]
```diff
@@ -598,7 +598,91 @@ LFAI = {
...
+			}
+		});
+
+		// The outermost div was initially styled with "display:none;" in order to prevent blinking when Alloy's
+		// JavaScript attempts to hide the div. At this point in JavaScript execution, Alloy is done manipulating the
+		// DOM and it is necessary to reset the style so that the component will be visible.
+		document.getElementById(clientId).style.display = null;
+	},
+
+	initAccordion: function(multiple, accordion, selectedIndexClientId, clientId, tabExpandedClientBehaviors,
```
Comment (ngriffin7a) [inline alloy/src/main/resources/META-INF/resources/liferay-faces-alloy/alloy.js:626]: @stiemannkj1 Functions that we add to alloy.js are technically API. If that is the case, then is there a convention we should use to indicate that a method is not API, such as starting a function name with an underscore?

**Comment by ngriffin7a** [comment]
bp to 2.x <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ngriffin7a** [comment]
Merged, thank you. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: alloy/src/main/resources/META-INF/resources/liferay-faces-alloy/alloy.js (hunk 1 above)
Comment (stiemannkj1) [reply alloy/src/main/resources/META-INF/resources/liferay-faces-alloy/alloy.js:626]: @ngriffin7a an underscore could be helpful, but I also thought that LFAI stood for Liferay Faces Alloy Internal.

Comment (ngriffin7a) [reply alloy/src/main/resources/META-INF/resources/liferay-faces-alloy/alloy.js:626]: Thanks, I was only looking at the diff -- I didn't scroll up enough in the plain text of the entire file to see that it was contained inside the LFAI namespace.

**Pull request merged**